packages/react-instantsearch/src/connectors/connectGeoSearch.js

```diff
@@ -0,0 +1,228 @@
+import createConnector from '../core/createConnector';
+import {
+  getResults,
+  getCurrentRefinementValue,
+  getIndex,
+  refineValue,
+  cleanUpValue,
+} from '../core/indexUtils';
+
+/**
+ * The GeoSearch connector provides the logic to build a widget that will display the results on a map.
+ * It also provides a way to search for results based on their position. The connector provides function to manage the search experience (search on map interaction).
+ * @name connectGeoSearch
+ * @kind connector
+ * @requirements Note that the GeoSearch connector uses the [geosearch](https://www.algolia.com/doc/guides/searching/geo-search) capabilities of Algolia.
+ * Your hits **must** have a `_geoloc` attribute in order to be passed to the rendering function. Currently, the feature is not compatible with multiple values in the `_geoloc` attribute
+ * (e.g. a restaurant with multiple locations). In that case you can duplicate your records and use the [distinct](https://www.algolia.com/doc/guides/ranking/distinct) feature of Algolia to only retrieve unique results.
+ * @propType {{ northEast: { lat: number, lng: number }, southWest: { lat: number, lng: number } }} [defaultRefinement] - Default search state of the widget containing the bounds for the map
+ * @providedPropType {function({ northEast: { lat: number, lng: number }, southWest: { lat: number, lng: number } })} refine - a function to toggle the refinement
+ * @providedPropType {function} createURL - a function to generate a URL for the corresponding search state
+ * @providedPropType {array.<object>} hits - the records that matched the search
+ * @providedPropType {boolean} isRefinedWithMap - true if the current refinement is set with the map bounds
+ * @providedPropType {{ northEast: { lat: number, lng: number }, southWest: { lat: number, lng: number } }} [currentRefinement] - the refinement currently applied
+ * @providedPropType {{ lat: number, lng: number }} [position] - the position of the search
+ */
+
+// To control the map with an external widget the other widget
+// **must** write the value in the attribute `aroundLatLng`
+const getBoundingBoxId = () => 'boundingBox';
+const getAroundLatLngId = () => 'aroundLatLng';
+const getConfigureAroundLatLngId = () => 'configure.aroundLatLng';
+
+const currentRefinementToString = currentRefinement =>
+  [
+    currentRefinement.northEast.lat,
+    currentRefinement.northEast.lng,
+    currentRefinement.southWest.lat,
+    currentRefinement.southWest.lng,
+  ].join();
+
+const stringToCurrentRefinement = value => {
+  const values = value.split(',');
+
+  return {
+    northEast: {
+      lat: parseFloat(values[0]),
+      lng: parseFloat(values[1]),
+    },
+    southWest: {
+      lat: parseFloat(values[2]),
+      lng: parseFloat(values[3]),
+    },
+  };
+};
+
+const latLngRegExp = /^(-?\d+(?:\.\d+)?),\s*(-?\d+(?:\.\d+)?)$/;
```

Comment:
Just checking, why the need for a different algorithm to parse the lat lng (regex vs split)?